src/Pelagos/Bundle/AppBundle/Command/DoiComparisonCommand.php

```diff
@@ -135,6 +145,25 @@ private function getResourceType(array $types): string
         return $resourceType;
     }
 
+    /**
+     * Checks if the array is empty.
+     *
+     * @param array  $doiMetadataElementArray Metadata element array to check if empty and key exists.
+     * @param string $keySearch               The key which needs to be checked if it exists in the array.
+     *
+     * @return boolean
+     */
+    private function doesKeyExist(array $doiMetadataElementArray, $keySearch): bool
```

Comment:
Cool, I like this a lot better. Now there is no doubt that the value/key we need is there.